albumentations/augmentations/functional.py

```diff
@@ -765,10 +765,10 @@ def add_rain(
         )
 
     image = cv2.blur(image, (blur_value, blur_value))  # rainy view are blurry
-    image_hls = cv2.cvtColor(image, cv2.COLOR_RGB2HLS).astype(np.float32)
-    image_hls[:, :, 1] *= brightness_coefficient
```

Comment:
It mught be better to still use HLS, but clip values to `[0, 255]` range. I think, that we can catch the same error if we will use HSV format without clipping.
```python
image_hls[:, :, 1] *= brightness_coefficient
image_hls[:, :, 1] = np.clip(image_hls[:, :, 1], 0, 255)
```

Reply:
I tested now and got the same corruptions as before.
![test2](https://user-images.githubusercontent.com/55686577/129720292-9185643d-3bde-46d3-9057-1ae3be8804b4.jpeg)
```  
    image_hls = cv2.cvtColor(image, cv2.COLOR_RGB2HLS).astype(np.float32)
    image_hls[:, :, 1] *= brightness_coefficient
    image_hls[:, :, 1] = np.clip(image_hls[:, :, 1], 0, 255)

    image_rgb = cv2.cvtColor(image_hls.astype(np.uint8), cv2.COLOR_HLS2RGB)```

Reply:
Naturally we should not be able to get neither overflow nor underflow errors, as the brightness coefficient should be between 0 and 1. The value cannot increase above 255 as we scale by a value below 1, and it cannot go below 0 as we scale by a non-negative value.

Therefore I do not believe clipping will solve this problem, and I also do not believe overflow is the issue.

Reply:
Ok, I got it. I see that in HLS colors have changed too, but in HSV they look the same.

> brightness coefficient should be between 0 and 1
Good point, I forgot that the transform doesn't support values greater than 1 for brightness coefficient.